### ISSUE TITLE
[Bug] error: #error C++17 or later compatible compiler is required to use ATen.  

### ISSUE THREAD

**Issue by Mxbonn** [open-mmlab/mmcv#2916]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmcv/issues) and [Discussions](https://github.com/open-mmlab/mmcv/discussions) but cannot get the expected help.
- [X] The bug has not been fixed in the latest version(https://github.com/open-mmlab/mmcv).

### Environment

N/A as problem is with installation

### Reproduces the problem - code sample

N/A -> installation issue

### Reproduces the problem - command or script

Use latest NVIDIA pytorch container (https://catalog.ngc.nvidia.com/orgs/nvidia/containers/pytorch): 23-08
`pip install openmim`
`mim install mmcv`

### Reproduces the problem - error message

[log.txt](https://github.com/open-mmlab/mmcv/files/12475232/log.txt)
TL;DR:  installation fails due to multiple `error: #error C++17 or later compatible compiler is required to use ATen.` errors

**Comment by Mxbonn** [comment]
Seems related to #2860
